common/config/keys_tree.go

```diff
@@ -240,7 +240,9 @@ func (tree *Tree) MergeWithKey(key string, obj interface{}, getValue func(val in
 					tree.MergeWithKey(subKey, elem.Field(i).Addr().Interface(), getValue)
 				} else if elem.Field(i).Type().Kind() == reflect.Ptr {
 					subKey := key + "." + idx
-					tree.MergeWithKey(subKey, elem.Field(i).Elem().Addr().Interface(), getValue)
+					if !elem.Field(i).IsNil() {
```

Comment:
Will this work while we merge config/cli/env ? like in config if it was not there then tree will not have the element. If we merge with condition that it should have been there that might fail.

Reply:
Shouldnt affect anything except pointer fields

Reply:
Regardless if the Field is Nil, the call to Addr() will fail